ee/ui/assets/generator/generator.go

```diff
@@ -0,0 +1,238 @@
+	)
+	if err := flagset.Parse(os.Args[1:]); err != nil {
+		level.Error(logger).Log("msg", "error parsing flags", "err", err)
+		os.Exit(1)
+	}
+
+	// relevel with the debug flag
+	logger = logutil.NewCLILogger(*flDebug)
+
+	missingOpt := false
+	for f, val := range map[string]string{
+		"out": *flOutdir,
+		"in":  *flIndir,
+	} {
+		if val == "" {
+			level.Error(logger).Log("msg", "Missing required flag", "flag", f)
+			missingOpt = true
+		}
+	}
+	if missingOpt {
+		os.Exit(1)
+	}
+
+	inDir = *flIndir
+	outDir = *flOutdir
+
+	// Find input icon names
+	iconNames := make(map[string]bool)
+	files, err := filepath.Glob(inDir + "/*")
+	if err != nil {
+		level.Error(logger).Log("msg", "error globbing input files", "error", err)
+		os.Exit(1)
+	}
+	for _, file := range files {
+		file = filepath.Base(file)
+		file = strings.TrimSuffix(file, ".png")
+		file = strings.TrimSuffix(file, ".svg")
+		iconNames[file] = true
+	}
+
+	if dir, err := os.MkdirTemp("", "icon-generator"); err != nil {
+		level.Error(logger).Log("msg", "error making tmpdir", "err", err)
+		os.Exit(1)
+	} else {
+		tmpDir = dir
+	}
+
+	ctx := context.Background()
+
+	for name, _ := range iconNames {
+		if err := generateIco(ctx, logger, name); err != nil {
+			level.Error(logger).Log(
+				"msg", "error generating ico",
+				"name", name,
+				"err", err)
+			os.Exit(1)
+		}
+
+		if err := generatePng(ctx, logger, name); err != nil {
+			level.Error(logger).Log(
+				"msg", "error generating png",
+				"name", name,
+				"err", err)
+			os.Exit(1)
+		}
+	}
+
+	if err := generateAssetGo(ctx, logger); err != nil {
+		level.Error(logger).Log("msg", "error expanding template", "error", err)
+		os.Exit(1)
+	}
+
+}
+
+func generateAssetGo(ctx context.Context, logger log.Logger) error {
+	output, err := os.Create(fmt.Sprintf("%s/assets.go", outDir))
+	if err != nil {
+		return fmt.Errorf("creating output file: %w", err)
+	}
+	defer output.Close()
+
+	embeds := make(map[string]string, len(embeddedFiles))
+	for _, filename := range embeddedFiles {
+		// embeds don't support a directory path. Everything is just in the outDir.
+		filename = filepath.Base(filename)
+		embeds[constName(filename)] = filename
+	}
+
+	tmpl, err := template.ParseFiles("generator/assets.go.tmpl")
+	if err != nil {
+		return fmt.Errorf("loading template: %w", err)
+	}
+
+	return tmpl.Execute(output, embeds)
+}
+
+// generatePng generates the png file.
+func generatePng(ctx context.Context, logger log.Logger, name string) error {
+	input := fmt.Sprintf("%s/%s.png", inDir, name)
+	output := fmt.Sprintf("%s/%s.png", outDir, name)
+
+	// append output. If we throw an error, we're going to ignore it anyhow.
+	embeddedFiles = append(embeddedFiles, output)
+
+	if skip, err := skipFile(input, output); err != nil {
+		return err
+	} else if skip {
+		level.Debug(logger).Log("msg", "skipping", "input", input, "output", output)
+		return nil
+	}
+
+	// FIXME Does this need scaling?
```

Comment:
Confusingly, I can't find Apple official guidance on sizes. Some unofficial sources say macOS menu bar icons should be max 22x22, ideally 16x16? Seems kind of small... I believe macOS will scale larger icons down to proper scale regardless 🤷 

https://iconhandbook.co.uk/reference/chart/osx/
https://bjango.com/articles/designingmenubarextras/

Reply:
I think it's better for us us to scale, less space in the binary

Reply:
Scaling to 22x22 didn't change sizes. So I removed the FIXME, and left it as a `cp`